feat(core): extract platforms out of core

vsavkin opened the pull request:
Currently, core depends on the browser, which means that other platforms (e.g., NativeScript or webworker) cannot use the bootstrapping logic core provides.
This PR extract makes bootstrapping logic in core completely platform-independent. The browser-specific code was moved to "angular2/platforms/browser".

BREAKING CHANGE

A few private helpers (e.g., platformCommon or applicationCommon) were removed or replaced with other helpers. Look at PLATFORM_COMMON_PROVIDERS, APPLICATION_COMMON_PROVIDERS, BROWSER_PROVIDERS, BROWSER_APP_PROVIDERS to see if they export the providers you need.

Closes #5219

vsavkin commented:
@IgorMinar reviewed it.

mary-poppins commented:
Merging PR #5280 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-5280](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-5280).

vsavkin committed:
feat(core): extract platforms out of core

Currently, core depends on the browser, which means that other platforms (e.g., NativeScript or webworker) cannot use the bootstrapping logic core provides.
This PR extract makes bootstrapping logic in core completely platform-independent. The browser-specific code was moved to "angular2/platforms/browser".

BREAKING CHANGE

A few private helpers (e.g., platformCommon or applicationCommon) were removed or replaced with other helpers. Look at PLATFORM_COMMON_PROVIDERS, APPLICATION_COMMON_PROVIDERS, BROWSER_PROVIDERS, BROWSER_APP_PROVIDERS to see if they export the providers you need.

Closes #5219

mary-poppins commented:
Merging PR #5280 on behalf of @vsavkin to branch [presubmit-vsavkin-pr-5280](https://github.com/angular/angular/tree/presubmit-vsavkin-pr-5280).
